8Knot/pages/contributors/visualizations/contribs_by_action.py

```diff
@@ -154,16 +166,31 @@ def toggle_popover(n, is_open):
     background=True,
 )
 def contribs_by_action_graph(repolist, interval, action):
-
     # wait for data to asynchronously download and become available.
-    cache = cm()
-    df = cache.grabm(func=ctq, repos=repolist)
-    while df is None:
-        time.sleep(1.0)
-        df = cache.grabm(func=ctq, repos=repolist)
+    while not_cached := cf.get_uncached(func_name=ctq.__name__, repolist=repolist):
+        logging.warning(f"{VIZ_ID}- WAITING ON DATA TO BECOME AVAILABLE")
+        time.sleep(0.5)
 
+    logging.warning(f"{VIZ_ID} - START")
     start = time.perf_counter()
-    logging.warning(f"{VIZ_ID}- START")
+
+    # GET ALL DATA FROM POSTGRES CACHE
+    df = cf.retrieve_from_cache(
+        VIZ_ID,
+        tablename="contributors",
+        repolist=repolist,
+        column_override=[
+            "id",
+            "repo_name",
+            "cntrb_id",
+            "created_at",
+            "login",
+            "action",
+            "rank",
+        ],
+    )
+
+    df = preproc_utils.contributors_df_action_naming(df)
```

Comment:
I see what you are doing here and need to figure out a good way to document this on the query and visualization side for devs know that it exists, how to use it, and how to add pandas query preprocessing there

Reply:
++ good point